File: day8/puzzle2.py
import re

chars = 0
mem = 0

def repl(p):
 return {
  '"': lambda: '\\"',
  '\\': lambda: '\\\\',
 }[filter(lambda x: x is not None, [p.group(0), p.group(1)])[0]]()

for i in open('day8.txt'):
 line = i.strip()
 chars += len(line)
 s = re.sub(r'(?:(\")|(\\))', repl, line, flags=re.I)
 mem += len(s) + 2
print mem - chars

File: day8/puzzle2.js
var fs = require('fs');

fs.readFile('day8.txt', 'utf-8', function (err, inp) {
 if (err) throw err;

 var chars = 0, mem = 0;
 inp.trim().split(/\n/g).forEach(function (line) {
  chars += line.length;
  mem += line.replace(/(\"|\\)/gi, function (a) { return "\\" + a; }).length + 2;
 });
 console.log(mem - chars);
});
